lib/model/rwfolder.go

```diff
@@ -715,6 +715,25 @@ func (p *rwFolder) deleteDir(file protocol.FileInfo) {
 			if defTempNamer.IsTemporary(file) {
 				osutil.InWritableDir(osutil.Remove, filepath.Join(realName, file))
 			}
+			if ignores.Nuke(file) {
+				if info, err := os.Stat(file); err == nil && info.IsDir() {
+					// Recursively check whether we can safely nuke the directory
+					err := filepath.Walk(file, func(path string, f os.FileInfo, err error) error {
+						if err != nil || !ignores.Nuke(path) {
+							return errors.New("Cannot nuke dir")
+						}
+						return nil
+					})
+					if err == nil {
+						filepath.Walk(file, func(path string, f os.FileInfo, err error) error {
```

Comment:
Files may be added between us walking the dir the first time and the second, files that we should not have removed. Also it would be nice if we cleaned out the files that we *should* remove and left the rest for manual cleanup. Perhaps do the deletes of files directly during the walk and queue the directories, or something?